x-pack/plugins/reporting/server/core.ts

```diff
@@ -239,12 +241,32 @@ export class ReportingCore {
 
   /**
    * If xpack.reporting.roles.enabled === true, register Reporting as a feature
-   * that is controlled by user role names
+   * that is controlled by user role names. Also, for Serverless register a
+   * 'shell' Reporting Kibana feature.
    */
-  public registerFeature() {
+  public registerFeatures({ isServerless }: { isServerless: boolean }) {
     const { features } = this.getPluginSetupDeps();
     const deprecatedRoles = this.getDeprecatedAllowedRoles();
 
+    // Register a 'shell' feature specifically for Serverless. If granted, it will automatically provide access to
+    // reporting capabilities in other features, such as Discover, Dashboards, and Visualizations. On its own, this
+    // feature doesn't grant any additional privileges.
+    if (isServerless) {
+      features.registerKibanaFeature({
+        id: 'reporting',
+        name: i18n.translate('xpack.reporting.features.reportingFeatureName', {
+          defaultMessage: 'Reporting',
+        }),
+        category: DEFAULT_APP_CATEGORIES.management,
+        app: [],
+        privileges: {
+          all: { savedObject: { all: [], read: [] }, ui: [] },
+          // No read-only mode currently supported
+          read: { disabled: true, savedObject: { all: [], read: [] }, ui: [] },
+        },
+      });
+    }
+
```

Comment:
@kertal, in some logic below, it looks like `features.enableReportingUiCapabilities();` will still be called in Serverless, which causes OSS features to declare Reporting sub-feature privileges. Would that create a problem?